app/frontend/src/components/designer/FormViewer.vue

```diff
@@ -354,27 +339,69 @@ export default {
           await new Promise((resolve) => setTimeout(resolve, 500));
         }
         if (this.confirmSubmit) {
-          this.doSubmit(submission, next);
+          this.confirmSubmit = false; // clear for next attempt
+          next();
+        } else {
+          // Force re-render form.io to reset submit button state
+          this.reRenderFormIo += 1;
         }
-        // Force re-render form.io to reset submit button state
-        this.reRenderFormIo += 1;
       } else {
-        this.doSubmit(submission, next);
+        next();
       }
     },
 
+    // FormIO submit event
     // eslint-disable-next-line no-unused-vars
     async onSubmit(submission) {
       if (this.preview) {
         alert('Submission disabled during form preview');
         return;
       }
+
+      const errors = await this.doSubmit(submission);
+
       // if we are here, the submission has been saved to our db
       // the passed in submission is the formio submission, not our db persisted submission record...
       // fire off the submitDone event.
       // console.info(`onSubmit(${JSON.stringify(submission)})`) ; // eslint-disable-line no-console
-      this.currentForm.events.emit('formio.submitDone');
+      if (errors) {
+        this.addNotification({
+          message: errors,
+          consoleError: `Error submiting the form: ${errors}`,
+        });
+      } else {
+        this.currentForm.events.emit('formio.submitDone');
+      }
+    },
+
+    // Not a formIO event, our saving routine to POST the submission to our API
+    async doSubmit(submission) {
+      // console.info(`doSubmit(${JSON.stringify(submission)})`) ; // eslint-disable-line no-console
```

Comment:
did you want to remove maybe?

Reply:
The commented console statements? Eh, there's a bunch in there and they're useful, so had just left the legacy comments in in the past honor of Jason :) 
We could go and chop them out probably later